relay-config/src/config.rs

```diff
@@ -87,6 +116,34 @@ pub enum ConfigErrorKind {
     /// compiled without the processing feature.
     #[fail(display = "was not compiled with processing, cannot enable processing")]
     ProcessingNotAvailable,
+    /// An override from a environment variable or a command line arg could not be
+    /// parsed into the proper type
+    #[fail(display = "Invalid override")]
+    InvalidOverride,
+}
+
+/// Structure used to hold information about configuration overrides via
+/// CLI parameters or environment variables
+#[derive(Debug, Default)]
+pub struct OverridableConfig {
+    /// the upstream for the Relay
+    pub upstream: Option<String>,
+    /// the host
+    pub host: Option<String>,
+    /// the port serving requests
+    pub port: Option<String>,
+    /// "true" if processing is enabled "false" otherwise
+    pub processing: Option<String>,
```

Comment:
Would it be possible to use an actual `bool` here?

Reply:
It seems that this is also used for the env var, which is passed directly in. In that case, this of course needs to stay a string.

Reply:
Initially it was a bool, and port was an i16 but I decided finally to make them all Option<String> in order to group all conversions in one place (i.e. in apply_override) instead of doing the conversions (and especially dealing with the errors) in both environment variables and parameters overrides.